code/modules/antagonists/traitor/objectives/kidnapping.dm

```diff
@@ -0,0 +1,294 @@
+	var/list/possible_areas = GLOB.the_station_areas.Copy()
+	for(var/area/possible_area as anything in possible_areas)
+		if(istype(possible_area, /area/station/hallway) || istype(possible_area, /area/station/security) || initial(possible_area.outdoors))
+			possible_areas -= possible_area
+
+	dropoff_area = pick(possible_areas)
+	replace_in_name("%TARGET%", target_mind.name)
+	replace_in_name("%JOB TITLE%", target_mind.assigned_role.title)
+	replace_in_name("%AREA%", initial(dropoff_area.name))
+	return TRUE
+
+/datum/traitor_objective/kidnapping/ungenerate_objective()
+	victim = null
+	dropoff_area = null
+
+/datum/traitor_objective/kidnapping/generate_ui_buttons(mob/user)
+	var/list/buttons = list()
+	if(!pod_called)
+		buttons += add_ui_button("Call Extraction Pod", "Pressing this will call down an extraction pod.", "rocket", "call_pod")
+	return buttons
+
+/datum/traitor_objective/kidnapping/ui_perform_action(mob/living/user, action)
+	. = ..()
+	switch(action)
+		if("call_pod")
+			if(pod_called)
+				return
+			var/area/user_area = get_area(user)
+			var/area/victim_area = get_area(victim)
+
+			if(user_area.type != dropoff_area)
+				to_chat(user, span_warning("You must be in [initial(dropoff_area.name)] to call the extraction pod."))
+				return
+
+			if(victim_area.type != dropoff_area)
+				to_chat(user, span_warning("[victim.real_name] must be in [initial(dropoff_area.name)] for you to call the extraction pod."))
+				return
+
+			call_pod(user)
+
+/datum/traitor_objective/kidnapping/proc/call_pod(mob/living/user)
+	pod_called = TRUE
+	var/obj/structure/closet/supplypod/extractionpod/new_pod = new()
+	RegisterSignal(new_pod, COMSIG_ATOM_ENTERED, .proc/enter_check)
+	new /obj/effect/pod_landingzone(get_turf(user), new_pod)
+
+/datum/traitor_objective/kidnapping/proc/enter_check(obj/structure/closet/supplypod/extractionpod/source, entered_atom)
+	if(!istype(source))
+		CRASH("Kidnapping objective's enter_check called with source being not an extraction pod: [source ? source.type : "N/A"]")
+
+	if(!ishuman(entered_atom))
+		return
+
+	var/mob/living/carbon/human/sent_mob = entered_atom
+
+	if(sent_mob.mind)
+		ADD_TRAIT(sent_mob.mind, TRAIT_HAS_BEEN_KIDNAPPED, TRAIT_GENERIC)
+
+	for(var/obj/item/belonging in sent_mob)
+		if(belonging == sent_mob.get_item_by_slot(ITEM_SLOT_ICLOTHING) || belonging == sent_mob.get_item_by_slot(ITEM_SLOT_FEET))
+			continue
+
+		sent_mob.transferItemToLoc(belonging)
+		victim_belogings.Add(belonging)
+
+	var/datum/bank_account/cargo_account = SSeconomy.get_dep_account(ACCOUNT_CAR)
+
+	if(cargo_account) //Just in case
+		cargo_account.adjust_money(-min(rand(1000, 3000), cargo_account.account_balance)) //Not so much, especially for competent cargo. Plus this can't be mass-triggered like it has been done with contractors
+
+	priority_announce("One of your crew was captured by a rival organisation - we've needed to pay their ransom to bring them back. \
```

Comment:
No backslash

If you want to multiline this correctly, put every arg on its own line